tests/SearchUtils.test.js

```diff
@@ -0,0 +1,149 @@
+        const { SearchUtils } = require('../js/search/searchUtils.js');
+        SearchUtilsAPI = new SearchUtils();
+        SearchUtilsAPI.path = userConfigDir;
+        done();
+    });
+
+    afterAll(function (done) {
+        fs.unlinkSync(searchConfig.FOLDERS_CONSTANTS.USER_CONFIG_FILE);
+        done();
+    });
+
+    describe('Tests for checking disk space', function () {
+
+        it('should return free sapce', function (done) {
+            const checkFreeSpace = jest.spyOn(SearchUtilsAPI, 'checkFreeSpace');
+            SearchUtilsAPI.checkFreeSpace().then(function () {
+                expect(checkFreeSpace).toHaveBeenCalled();
+                done();
+            });
+        });
+
+        it('should return error', function (done) {
+            const checkFreeSpace = jest.spyOn(SearchUtilsAPI, 'checkFreeSpace');
+            SearchUtilsAPI.path = undefined;
+            SearchUtilsAPI.checkFreeSpace().catch(function (err) {
+                expect(err).toEqual(new Error("Please provide path"));
+                expect(checkFreeSpace).toHaveBeenCalled();
+                done();
+            });
+        });
+
+        it('should return error invalid path', function (done) {
+            const checkFreeSpace = jest.spyOn(SearchUtilsAPI, 'checkFreeSpace');
+            SearchUtilsAPI.path = './tp';
+            SearchUtilsAPI.checkFreeSpace().catch(function (err) {
+                expect(checkFreeSpace).toHaveBeenCalled();
+                expect(err).toEqual(err);
+                done();
+            });
+        });
+    });
+
+    describe('Test for search users config', function () {
+
+        it('should return null for new user config', function (done) {
+            SearchUtilsAPI.getSearchUserConfig(1234567891011).then(function (res) {
+                expect(res).toEqual(null);
+                done();
+            });
+        });
+
+        it('should exist users config file', function (done) {
+            setTimeout(function () {
+                expect(fs.existsSync(searchConfig.FOLDERS_CONSTANTS.USER_CONFIG_FILE)).toEqual(true);
+                done();
+            }, 2000)
+        });
+
+        it('should exist users config file', function (done) {
+            setTimeout(function () {
+                SearchUtilsAPI.getSearchUserConfig(1234567891011).then(function (res) {
+                    expect(res).toEqual({});
+                    done();
+                });
+            }, 3000)
+        });
+
+        it('should update user config file', function (done) {
+            let data = {
+                rotationId: 0,
+                version: 1,
+                language: 'en'
+            };
+            SearchUtilsAPI.updateUserConfig(1234567891011, data).then(function (res) {
+                expect(res).toEqual(data);
+                done();
+            })
+        });
+
+        it('should modify user config file', function (done) {
```

Comment:
So, what is the difference between the above tests?

Reply:
The first test is to update the user data which will be empty (the rotation-id will be 0).
In the second test changing the rotation-id and updating to config (after the updated rotation-id will be 1).

Reply:
👍